sdk/src/main/java/com/karhoo/sdk/analytics/Payloader.kt

```diff
@@ -157,6 +238,101 @@ class Payloader internal constructor(builder: Builder) {
             return this
         }
 
+        fun checkoutOpened(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun cardAuthorisationSuccess(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun loyaltyStatusRequested(
+            slug: String?,
+            quoteId: String?,
+            correlationId: String?,
+            loyaltyStatus: LoyaltyStatus?,
+            balance: Int?,
+            loyaltyProgramme: LoyaltyProgramme?
+        ): Builder {
+            payload[LOYALTY_ENABLED] =
+                loyaltyStatus?.canBurn == false && loyaltyStatus.canEarn == false
+            payload[LOYALTY_STATUS_SUCCESS] = loyaltyStatus != null
+            payload[LOYALTY_STATUS_CAN_BURN] = loyaltyStatus?.canBurn ?: false
+            payload[LOYALTY_STATUS_CAN_EARN] = loyaltyStatus?.canEarn ?: false
+            payload[LOYALTY_STATUS_BALANCE] = balance ?: 0
```

Comment:
I don't think that default values are correct approach. Ballance == null is not the same as ballance == 0, same fo canEarn/burn statuses. If loyaltyStatus doesn't exist then we should send LOYALTY_STATUS_SUCCESS = false and don't send LOYALTY_STATUS_CAN_BURN, LOYALTY_STATUS_CAN_EARN and LOYALTY_STATUS_BALANCE